gammapy/cube/make.py

```diff
@@ -33,196 +33,313 @@ class MapMakerObs:
-            bkg=self.observation.bkg,
-            geom=self.geom,
+            pointing=pointing,
+            ontime=observation.observation_time_duration,
+            bkg=observation.bkg,
+            geom=geom,
             oversampling=self.background_oversampling,
         )
-        if self.fov_mask is not None:
-            background.data[..., self.fov_mask] = 0
+        return background
 
-        # TODO: decide what background modeling options to support
-        # Extra things like FOV norm scale or ring would go here.
+    def make_edisp(self, observation):
+        """Make edisp map.
 
-        self.maps["background"] = background
+        Parameters
+        ----------
+        observation : `DataStoreObservation`
+            Observation container.
 
-    def _make_edisp(self):
-        energy_axis = self.geom_true.get_axis_by_name("ENERGY")
-        geom_migra = self.geom_true.to_image().to_cube([self.migra_axis, energy_axis])
-        edisp_map = make_edisp_map(
-            edisp=self.observation.edisp,
-            pointing=self.observation.pointing_radec,
-            geom=geom_migra,
+        Returns
+        -------
+        edisp : `EdispMap`
+            Edisp map.
+        """
+        geom = self._cutout_geom(self.geom_edisp, observation)
+
+        exposure = self.make_exposure_irf(observation)
+
+        edisp = make_edisp_map(
+            edisp=observation.edisp,
+            pointing=observation.pointing_radec,
+            geom=geom,
             max_offset=self.offset_max,
-            exposure_map=self.maps["exposure_irf"],
+            exposure_map=exposure,
         )
-        self.maps["edisp"] = edisp_map
+        return edisp
+
+    def make_psf(self, observation):
+        """Make psf map.
+
+        Parameters
+        ----------
+        observation : `DataStoreObservation`
+            Observation container.
+
+        Returns
+        -------
+        psf : `PSFMap`
+            Psf map.
+        """
+        psf = observation.psf
+        geom = self._cutout_geom(self.geom_psf, observation)
 
-    def _make_psf(self):
-        psf = self.observation.psf
         if isinstance(psf, EnergyDependentMultiGaussPSF):
             psf = psf.to_psf3d()
-        energy_axis = self.geom_true.get_axis_by_name("ENERGY")
 
-        geom_rad = self.geom_true.to_image().to_cube([self.rad_axis, energy_axis])
-        psf_map = make_psf_map(
+        exposure = self.make_exposure_irf(observation)
+
+        psf = make_psf_map(
             psf=psf,
-            pointing=self.observation.pointing_radec,
-            geom=geom_rad,
+            pointing=observation.pointing_radec,
+            geom=geom,
             max_offset=self.offset_max,
-            exposure_map=self.maps["exposure_irf"],
+            exposure_map=exposure,
         )
-        self.maps["psf"] = psf_map
+        return psf
+
+    @lru_cache(maxsize=1)
+    def make_mask_safe(self, observation):
```

Comment:
OK. See the comment above regarding when to make the safe range.

Reply:
Will be addressed in a follow-up PR...